lib/mayaUsd/utils/util.cpp

```diff
@@ -226,6 +228,20 @@ MStatus UsdMayaUtil::GetDagPathByName(const std::string& nodeName, MDagPath& dag
     return status;
 }
 
+UsdStageRefPtr UsdMayaUtil::GetStageByProxyName(const std::string& proxyPath)
```

Comment:
I get that this is to remove dependence on UFE, but in doing so this is adding a by-name lookup of a proxy shape.  If it's not called a lot, this is fine.

Also, I think it would be good to steer readers of this code to lib/mayaUsd/ufe/Utils.h, which will have much better performance, if a dependence on UFE is not an issue.

Reply:
Couple thoughts:
1. Do you think lookup performance would be bad given a unique path?  It might not be as fast as some of the structures in UFE, but I would hope that when looking for a single object given a unique string that it would be fairly quick.  Right now this function will be called once per proxy shape when saving, so it shouldn't be too many and definitely not called interactively.  

2. We are going to have to revisit the utility function that I am replacing soon anyways, since it will no longer be valid to map between proxies and stages like we have it now.  The stage cache map code that can go from UsdStage ptr to proxy is the one that will have to change.

Reply:
Once per proxy shape when saving is fine, not an issue.